src/charm.py

```diff
@@ -53,6 +55,7 @@ def __init__(self, *args):
         super().__init__(*args)
         self._bessd_container_name = self._bessd_service_name = "bessd"
         self._routectl_service_name = "routectl"
+        self._restart_pod = False
```

Comment:
Similar to my other comment, let's not have/set a global value for restart pod. Hooks that need to know this should call `_is_pod_restart_required` on their own instead of looking at this value. We shouldn't need `self._restart_pod` anywhere here.

Reply:
There is a reason to do store the boolean in a variable before running it under config changed after new NADs are created. 
KubernetesMultusCharmLib library run _configure_multus method which deletes the old NAD definitions and creates new NAD definitions with latest config settings when config_changed event is fired. In config_changed method of charm if we try to detect whether pod restart is required or not, it always returns False. So this check needs to be done before config_changed run and restart needs to be done after new NADs are created to make the new NADs take effect.

Reply:
Pod restart  needs to wait till new NADs are created and pod restart detection should be done before old NADs deleted.  I was simply storing the status to take the action later. I am open to any other suggestions to overcome this situation.